config/manager/manager.yaml

```diff
@@ -31,6 +31,28 @@ spec:
         # versions < 1.20 or on vendors versions which do NOT support this field by default (i.e. Openshift < 4.11 ).
         # seccompProfile:
         #   type: RuntimeDefault
+      initContainers:
+      - name: check-ray-crds
+        securityContext:
+          allowPrivilegeEscalation: false
+          capabilities:
+            drop:
+              - "ALL"
+        image: $(codeflare_operator_controller_image)
+        command:
+        - sh
+        - -c
+        - |
+          set -e
+          CRDS="rayclusters.ray.io rayjobs.ray.io rayservices.ray.io"
+          for crd in $CRDS; do
+            echo "Checking for $crd"
+            until oc get crd $crd; do
+              echo "$crd not available yet, retrying in 10 seconds..."
+              sleep 10
```

Comment:
what happens CRD is not installed, not just not available for short period? 
this will cause initContainer never stop?

Reply:
That would be the current behaviour.  The CFO shouldn't start without those CRDs previously installed, hence, the initContainer will continue to search for them until they are present.

Should I have a timeout for it? In case of timing out, I suppose the only way of 're-activating' the CFO would be to restart the pod.

Reply:
so if the bash script timeout, it should exit non-zero, which by default will cause Pod restart without start the "real" container.
i think this can be easily verified in the cluster.